include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -985,40 +992,57 @@ struct __parallel_find_backward_tag
 #else
     using _AtomicType = oneapi::dpl::__internal::__difference_t<_RangeType>;
 #endif
-    using _Compare = oneapi::dpl::__internal::__pstl_greater;
 
     template <typename _DiffType>
     constexpr static _AtomicType __init_value(_DiffType)
     {
         return _AtomicType{-1};
     }
+
+    template <typename _TAtomic>
+    static void
+    __save_state_to_atomic(_TAtomic& __found, _AtomicType __new_state)
+    {
+        // As far as we make search from end to the begin of data, we should save the last (maximal) found state.
+        __found.fetch_max(__new_state);
+    }
 };
 
 // Tag for __parallel_find_or for or-semantic
 struct __parallel_or_tag
 {
-    class __atomic_compare
-    {
-      public:
-        template <typename _LocalAtomic, typename _GlobalAtomic>
-        bool
-        operator()(const _LocalAtomic& __found_local, const _GlobalAtomic& __found) const
-        {
-            return __found_local == 1 && __found == 0;
-        }
-    };
-
     using _AtomicType = int32_t;
```

Comment:
Should this use the same Atomic type as in `__parallel_find_[backward,forward]_tag`?

Follow ups:  
1) Does `backward` have a reason to be signed int rather than unsigned?
2) Do we check that `n` is less than max (index type)?  (perhaps we have other problems with `n` > 2 Billion)

Reply:
About "Should this use the same Atomic type as in __parallel_find_[backward,forward]_tag?"
 - it's simple not required because we save the fact that we found something, not real position. From this point of view it may be `bool` type too. But I assume that we haven't any additional perf proof from that.

Reply:
Yes, ok that is right for the `or` case.  However, I think it cant be bool because atomics must be at least 32bits.  

I think my follow ups still make sense for `__parallel_find_[backward,forward]_tag` though.  
Should we align these with each other?